### PR TITLE
Change tags to list

### PR DESCRIPTION
**Pull request by ondrae**
Working on https://github.com/codeforamerica/brigade/issues/374

#### Tasks
- [x] Write tests for tags as a list
- [x] Write docs for tags as list
- [x] Write code for tags as list
- [x] Merge master back into this PR
- [x] Edit migration file to work after #283

### CODE REVIEW

**Comment by ondrae** [comment]
@tmaybe This is ready for your review.

After it is merged, we'll need to run these
```
heroku run python app.py db upgrade --app cfapi-staging
heroku run python app.py db upgrade --app codeforamerica-api
```

We'll probably need to ensure that anything with existing tags get reupdated.
```
heroku pg:psql --app cfapi-staging
UPDATE project SET last_updated = null WHERE tags IS NOT NULL;
\q
heroku pg:psql --app codeforamerica-api
UPDATE project SET last_updated = null WHERE tags IS NOT NULL;
\q
```

File: run_update.py [hunk 1]
```diff
@@ -347,6 +347,9 @@ def get_projects(organization):
                         # some values might be empty strings
                         elif type(project_value) in (str, unicode) and unicode(project_value.decode('utf8')) == u'':
                             project[project_key] = None
+                        # we want tags to be a list with no whitespace
+                        elif project_key == 'tags':
+                            project[project_key] = [tag.strip() for tag in project_value.split(',')]
```
Comment (tmaybe) [inline run_update.py]: Won't lists get caught by the first condition? (`if type(project_value) is list`)

If it won't, we should encode the tags before we save them, maybe add `project_value = unicode(project_value.decode('utf8'))` before splitting it into the list.

Comment (tmaybe) [inline run_update.py]: And maybe exclude empty values by adding `if tag.strip() is not u''` to the list comprehension?

**Comment by ondrae** [comment]
Okay @tmaybe this is ready for you to look at again.

I've got better tests and better list making.

**Comment by tmaybe** [comment]
Looks good, merging. Let me know if you want me to deploy.

**Comment by ondrae** [comment]
We might need to run the below tags clearing code before the db migration. Let me check how staging works out on the next update.

**Comment by ondrae** [comment]
Nvm, works great after an update.

**Comment by ondrae** [comment]
Pushing to production

**Pull request merged**